app/src/main/java/com/example/fina/screen/detail/DetailPresenter.kt

```diff
@@ -0,0 +1,92 @@
+package com.example.fina.screen.detail
+
+import android.content.ContentValues.TAG
+import android.content.Context
+import android.graphics.Color
+import android.util.Log
+import androidx.core.content.ContextCompat
+import com.example.fina.R
+import com.example.fina.data.model.PriceRecord
+import com.example.fina.data.repository.CoinRepository
+import com.example.fina.data.repository.OnResultListener
+import com.example.fina.utils.ExtraParams
+import com.example.fina.utils.MarketCapInterval
+import com.github.mikephil.charting.data.Entry
+import com.github.mikephil.charting.data.LineData
+import com.github.mikephil.charting.data.LineDataSet
+
+class DetailPresenter internal constructor(private val coinRepository: CoinRepository?) :
```

Comment:
Repository không được phép null em
```suggestion
class DetailPresenter internal constructor(private val coinRepository: CoinRepository) :
```